hera_cal/frf.py

```diff
@@ -497,6 +498,85 @@ def filter_data(self, data, frps, flags=None, nsamples=None,
             filt_flags[k] = f
             filt_nsamples[k] = eff_nsamples
 
+    def run_tophat_frfilter(self, to_filter=None, weight_dict=None, mode='clean',
```

Comment:
So this is kind of just a wrapper around `VisClean.fourier_filter` with extra tools for handling sky_frate kwarsg. Can we make that more clear in the name and in docstring, just b/c there is now a proliferation of "filtering" functions that this subclass inherits. (I also think `run_` should be reserved for module-level functions used directly by drive scripts) Perhaps something like

```
def tophat_frfilter(self, keys=None, wgts=None, ...):
    A wrapper around VisClean.fourier_filter specifically for
    filtering along the time axis with uniform fringe-rate weighting.
````

also is there a reason why this is specifically tophat weighting? It seems it would be nice to allow
for arbitrary fr-weighting. but perhaps that is actually much harder when we are using the `fourier_filter`
function and all of its possible modes. I just raise this as it seems like this is inching towards the direction
that ARP brought of up optimal or semi-optimal FR filtering. it would be nice to be able to do this simultaneously with inpainting.

Reply:
I agree with the name change.  I've fixed up the docstring.


Regarding the tophat weighting, this is the only possible functionality I can imagine for `fourier_filter` since the filter fits modes that span a compact region of fringe-rate space to the data with modes that are not necessarily Fourier modes (and can be mapped to specific fringe-rates). For example, DPSS modes are not necessarily super localized to a specific fringe-rate (although they are super concentrated within the bandwidth of the filter). For fringe-rate (fourier) modes, we can up/down weight different modes but I think this is basically what `apply_fir` function does. 

I think the most sensible way to support non-tophat weights in the future is to use `apply_fir` on the clean components after running the `tophat_frfilter` to achieve different fringe-rate weights. This step can be added to the `load_tophat_frfilter_and_write` function and when we do this, we can also change the name to be something more general. The next PR  ( #736 ) has functions to calculate fringe-rate profiles from uvbeam objects so we can use this in conjunction with `frp_to_fir` and `apply_fir`.